src/modules/did/did.entity.ts

```diff
@@ -1,19 +1,177 @@
+import {
+  IServiceEndpoint,
+  IDIDDocument,
+  IAuthentication,
+  ILinkedDataProof,
+  IPublicKey,
+} from '@ew-did-registry/did-resolver-interface';
+import { Field, ID, Int, ObjectType } from '@nestjs/graphql';
 import { Column, Entity, PrimaryColumn } from 'typeorm';
+import { JSONObject } from '../../common/json.scalar';
+
+interface ProfileClaim {
+  profile?: any;
+}
+
+interface ENSClaim {
+  claimType?: string;
+  claimTypeVersion?: string;
+}
+
+export interface IClaim extends IServiceEndpoint, ProfileClaim, ENSClaim {}
+
+@ObjectType()
+class AuthenticationSchema implements IAuthentication {
+  @Field(() => Int, { nullable: true })
+  block?: number;
+
+  @Field()
+  publicKey: string;
+
+  @Field()
+  type: string;
+
+  @Field(() => Int)
+  validity: any;
+
+  [key: string]: string | number;
+}
+
+@ObjectType()
+class ProofSchema implements ILinkedDataProof {
+  @Field()
+  created: string;
+
+  @Field()
+  creator: string;
+
+  @Field()
+  signatureValue: string;
+
+  @Field()
+  type: string;
+}
+
+@ObjectType()
+class PublicKeySchema implements IPublicKey {
+  @Field(() => Int, { nullable: true })
+  block: number;
+
+  @Field()
+  controller: string;
+
+  @Field({ nullable: true })
+  ethereumAddress?: string;
+
+  @Field()
+  id: string;
+
+  @Field({ nullable: true })
+  publicKeyBase58?: string;
+
+  @Field({ nullable: true })
+  publicKeyBase64?: string;
+
+  @Field({ nullable: true })
+  publicKeyHex?: string;
+
+  @Field({ nullable: true })
+  publicKeyJwk?: string;
+
+  @Field({ nullable: true })
+  publicKeyMultibase?: string;
+
+  @Field({ nullable: true })
+  publicKeyPem: string;
+
+  @Field({ nullable: true })
+  type: string;
+
+  @Field(() => Int, { nullable: true })
+  validity: any;
+
+  [key: string]: string | number;
+}
+
+@ObjectType()
+class ClaimSchema implements IClaim {
+  @Field(() => Int, { nullable: true })
+  block: number;
+
+  @Field({ nullable: true })
+  claimType?: string;
+
+  @Field({ nullable: true })
+  claimTypeVersion?: string;
+
+  @Field()
+  description: string;
+
+  @Field()
+  hash: string;
+
+  @Field()
+  id: string;
+
+  @Field(() => JSONObject, { nullable: true })
+  profile?: any;
+
+  @Field()
+  serviceEndpoint: string;
+
+  @Field()
+  type: string;
+
+  @Field(() => Int)
+  validity: any;
+
+  [key: string]: string | number;
+}
```

Comment:
Interesting that all of the fields of the DIDDocument have been broken out. I'm on the fence as to whether or not this is a good idea. On one hand, it would be possible performing richer queries on the data. On the other hand, it adds to the complexity of the caching. I think if you've already implemented it we can use it but I'm wondering what are your thoughts are on why you decided to implement all of the DIDDocument properties?

Reply:
For purpose of the caching is not adding any complexity, cause we're storing almost all of this data as `jsonb` column and it will accept any kind of data, this is made just for graphql queries, so we'll be able to access deeply nested objects.